### ISSUE TITLE
After updating NixOS yesterday, the screen reader is automatically activated and cannot be turned off.

### ISSUE THREAD

**Issue by ares191166** [niri-wm/niri#2830]
NIRI After updating NixOS yesterday, the screen reader is automatically activated and cannot be turned off.

**Comment by HigherOrderLogic** [comment]
Just dont spawn Orca?

**Comment by ares191166** [comment]
> 就是不能生出奥卡?

The user's question is: "I didn't spawn Orca, and I can't turn off the screen reader with SUPER+ALT+S either."

**Comment by Saladin1812** [comment]
Ya same issue. even `pkill orca` doesn't seem to work !

**Comment by Saladin1812** [comment]
> Just dont spawn Orca?

Nothing in my config is spawning orca. It is something auto in the latest update. wierd but it is what it is.

**Comment by piotr152** [comment]
Same here, I was able to stop it via systemd:

```
systemctl --user stop orca.service
```

This seems to be caused by the recent Gnome update to v49.

```
❯ nvd diff /nix/var/nix/profiles/system-1215-link /run/current-system | grep orca
[U*]  #411  orca                                         48.6 -> 49.4
[A.]  #27  unit-orca.service                            <none>
```

I was able to deactivate it with

```nix
systemd.user.services.orca.enable = false;
```

So in summary - not  a niri issue, I think.

**Comment by Sempyos** [comment]
Probably not, if one of you can do ``systemctl --user status orca`` and see if it's enabled by default now, that would be helpful

**Comment by piotr152** [comment]
This is the output of the status command
```bash
~ ❯ systemctl status --user orca
● orca.service - Orca Screen Reader
     Loaded: loaded (/etc/systemd/user/orca.service; enabled; preset: ignored)
    Drop-In: /nix/store/p6q0yszll6k92gfz3pi5av6alyc56g7r-user-units/orca.service.d
             └─overrides.conf
     Active: active (running) since Tue 2025-11-18 23:58:25 CET; 35s ago
 Invocation: d49788bfa91a4e20aa52fa79f0afbdf8
       Docs: man:orca(1)
             https://orca.gnome.org/
   Main PID: 3788 (orca)
      Tasks: 14 (limit: 18913)
     Memory: 47.3M (peak: 49M)
        CPU: 2.706s
     CGroup: /user.slice/user-1000.slice/user@1000.service/session.slice/orca.service
             └─3788 orca

Nov 18 23:58:25 galactica systemd[2839]: orca.service: Scheduled restart job, restart counter is at 1.
Nov 18 23:58:25 galactica systemd[2839]: Starting Orca Screen Reader...
Nov 18 23:58:25 galactica orca[3788]: AT-SPI: Error in GetItems, sender=:1.0, error=Unknown object '/org/a11y/atspi>
Nov 18 23:58:25 galactica systemd[2839]: Started Orca Screen Reader.
Nov 18 23:58:42 galactica orca[3788]: /nix/store/ry43cb84lmpkbdi8c5kpw1bj84a2vc3b-orca-49.4/lib/python3.13/site-pac>
Nov 18 23:58:42 galactica orca[3788]:   self._contents = []
```

This is how the unit looks like:

```ini
[Unit]
Description=Orca Screen Reader
Documentation=man:orca(1)
Documentation=https://orca.gnome.org/

PartOf=graphical-session.target
After=graphical-session.target

# We're going to need this, so instead of waiting for it to
# dbus activate later, let's start it in parallel with Orca
Wants=at-spi-dbus-bus.service

[Service]
Type=notify-reload
ExecStart=/nix/store/ry43cb84lmpkbdi8c5kpw1bj84a2vc3b-orca-49.4/bin/orca --replace
Restart=always
WatchdogSec=6
Slice=session.slice

[Install]
WantedBy=graphical-session.target
```

Niri invokes the graphical-session target which in turn starts orca. It's a NixOS issue that can be worked around using the configuration line mentioned in my previous comment.

**Comment by Sempyos** [comment]
Just wanted to check, but ya, I don't think is a niri issue as you said

**Comment by Rhodwell-Chima** [comment]
I was able to solve this issue by adding the following line to `configuration.nix` file.
```nix
  services.orca.enable = false;
```
After that, when you run:
```bash
systemctl --user status orca
```
you should get the following output:

> 
> ```
> Unit orca.service could not be found.
> ```
Showing that its deactivated.

**Comment by Sempyos** [comment]
@ares191166 can we close this now..?

**Comment by RossSmyth** [comment]
This is bug upstream in Nixpkgs, not in Niri.
https://github.com/NixOS/nixpkgs/issues/462935

A fix will be hitting unstable pretty soon
https://nixpkgs-tracker.ocfox.me/?pr=463373

Until then it can be worked around with this line:
`systemd.user.services.orca.wantedBy = lib.mkForce []`

**Comment by uonr** [comment]
After updating to the latest unstable nixpkgs, the problem no longer appears.

**Comment by Sempyos** [comment]
Then I feel this can be closed then

**Comment by ares191166** [comment]
> 然后我觉得这个可以关闭

You can close it now, because this bug only occurred in niri at that time, while hyprland and gnome were working fine.

**Comment by Sempyos** [comment]
I don't have permissions here to close for others, I need you to close it sorry

**Comment by ares191166** [comment]
'你好。已收到你的邮件，我会在假期结束后尽快处理。————————————这是来自QQ邮箱的自动回复邮件。'
